tests/Backend/SOX/ExportTableTest.php

```diff
@@ -0,0 +1,171 @@
+
+        $projectManagerDevBranchClient = $this->getBranchAwareClient($newBranch['id'], [
+            'token' => STORAGE_API_DEFAULT_BRANCH_TOKEN,
+            'url' => STORAGE_API_URL,
+        ]);
+
+        $projectManagerDevBranchExporter = new TableExporter($projectManagerDevBranchClient);
+        try {
+            $projectManagerDevBranchExporter->exportTable($tableId, $this->downloadPath, $exportOptions);
+            $this->fail('Project manager cannot export from table in devBranch');
+        } catch (ClientException $e) {
+            $this->assertStringContainsString(
+                'You don\'t have access to the resource.',
+                $e->getMessage()
+            );
+        }
+
+        $defaultBranch = $this->branches->getDefaultBranch();
+        $developerDefaultBranchClient = $this->getBranchAwareClient($defaultBranch['id'], [
+            'token' => STORAGE_API_DEVELOPER_TOKEN,
+            'url' => STORAGE_API_URL,
+        ]);
+
+        $developerDefaultBranchExporter = new TableExporter($developerDefaultBranchClient);
+
+        try {
+            $developerDefaultBranchExporter->exportTable($tableId, $this->downloadPath, $exportOptions);
+            $this->fail('Cannot export from table in devBranch via default branch client');
+        } catch (ClientException $e) {
+            $this->assertStringContainsString(
+                sprintf('The table "languages" was not found in the bucket "%s" in the project ', $devBranchBucketId),
+                $e->getMessage()
+            );
+        }
+    }
+
+    /**
+     * @dataProvider tableExportData
+     */
+    public function testTableAsyncExportInDefaultBranch(CsvFile $importFile, string $expectationsFileName, array $exportOptions = []): void
+    {
+        $expectationsFile = __DIR__ . '/../../_data/' . $expectationsFileName;
+
+        $description = $this->generateDescriptionForTestObject();
+        $projectManagerDefaultBranchClient = $this->getDefaultBranchStorageApiClient();
+
+        $productionBucketId = $this->initEmptyBucket(
+            $this->getTestBucketName($description),
+            self::STAGE_IN,
+            $description,
+            $projectManagerDefaultBranchClient
+        );
+
+        $tableId = $projectManagerDefaultBranchClient->createTableAsync($productionBucketId, 'languages', $importFile);
+
+        $projectManagerDefaultBranchExporter = new TableExporter($projectManagerDefaultBranchClient);
+        $projectManagerDefaultBranchExporter->exportTable($tableId, $this->downloadPath, $exportOptions);
+
+        // compare data
+        $this->assertTrue(file_exists($this->downloadPath));
+        $this->assertLinesEqualsSorted(file_get_contents($expectationsFile), file_get_contents($this->downloadPath), 'imported data comparison');
+
+        $newBranch = $this->branches->createBranch($this->generateDescriptionForTestObject() . '_aaaa');
```

Comment:
tady by asi taky mohlo jit pouzit `createDevBranchForTestCase`

Reply:
to bohuzial nejde, ono to pouziva `$this->_client` aspadne to na tom ze nema pristup na resource